http3/roundtrip.go

```diff
@@ -278,10 +278,35 @@ func isNotToken(r rune) bool {
 	return !httpguts.IsTokenRune(r)
 }
 
+func resolveUDPAddrWithCtx(ctx context.Context, netstr string, addr string) (*net.UDPAddr, error) {
+	var (
+		err        error
+		host, port string
+		portnum    int
+	)
+	if host, port, err = net.SplitHostPort(addr); err != nil {
+		return nil, err
+	}
+	if portnum, err = net.DefaultResolver.LookupPort(ctx, netstr, port); err != nil {
```

Comment:
Why is this needed? Why not just use the `port`? Does it cause a potential DNS roundtrip?